### PR TITLE
Ported PR https://github.com/fabric8io/fabric8-maven-plugin/pull/1711

### PR DESCRIPTION
**Pull request by rohanKanojia**
+ Add a generator for Open Liberty that extends the java-exec generator. It defaults webPort to the Liberty default port (9080) and adds an environment variable for use by S2I if it detects a Liberty runnable jar.
+ Add Liberty's default web ports (9080, 9443) to the ExposeEnricher so that routes get generated.

I would add the sample once #33 gets merged

### CODE REVIEW

**Comment by rohanKanojia** [comment]
@manusa : Could you please review?

File: jkube-kit/jkube-kit-openliberty/pom.xml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+            <groupId>org.eclipse.jkube</groupId>
+            <artifactId>jkube-maven-enricher-api</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>org.eclipse.jkube</groupId>
+            <artifactId>jkube-maven-enricher-specific</artifactId>
+        </dependency>
+
+        <dependency>
```
Comment (manusa) [inline jkube-kit/jkube-kit-openliberty/pom.xml]: Can be removed, it's transitively included by `jkube-maven-generator-java-exec`

File: jkube-kit/jkube-kit-openliberty/pom.xml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+            <groupId>org.eclipse.jkube</groupId>
+            <artifactId>jkube-maven-generator-java-exec</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>org.eclipse.jkube</groupId>
+            <artifactId>jkube-kit-common</artifactId>
+        </dependency>
+
+        <dependency>
```
Comment (manusa) [inline jkube-kit/jkube-kit-openliberty/pom.xml]: Can be removed, it's transitively included by all of the other jkube dependencies

File: jkube-kit/jkube-kit-openliberty/pom.xml [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    <parent>
+        <artifactId>jkube-kit-parent</artifactId>
+        <groupId>org.eclipse.jkube</groupId>
+        <version>0.1.1-SNAPSHOT</version>
+        <relativePath>../parent/pom.xml</relativePath>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>jkube-kit-openliberty</artifactId>
+
```
Comment (manusa) [inline jkube-kit/jkube-kit-openliberty/pom.xml]: Please add the name entry:
`<name>Jkube Kit :: OpenLiberty</name>`

**Comment by manusa** [comment]
I'm also missing this one: https://github.com/brenthdaniel/fabric8-maven-plugin/blob/1436facaab80ff818b4151dc3b5b94541dac7c1b/generator/openliberty/src/test/java/io/fabric8/generator/openliberty/OpenLibertyGeneratorTest.java

File: CHANGELOG.md [hunk 4]
```diff
@@ -22,6 +22,7 @@ Usage:
 ```
 ### 0.2-SNAPSHOT
 * Fix #71: script to extract changelog information for notifications
+* Ported OpenLiberty support
```
Comment (manusa) [inline CHANGELOG.md]: Add link to current PR so changelog extractor script can generate a link
```suggestion
* #34 Ported OpenLiberty support
```

File: jkube-kit/jkube-kit-openliberty/pom.xml [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>jkube-kit-parent</artifactId>
+        <groupId>org.eclipse.jkube</groupId>
+        <version>0.2-SNAPSHOT</version>
+        <relativePath>../parent/pom.xml</relativePath>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+    <artifactId>jkube-kit-openliberty</artifactId>
+    <name>Jkube Kit :: Open Liberty</name>
```
Comment (manusa) [inline jkube-kit/jkube-kit-openliberty/pom.xml]: `s/Jkube/JKube/`

**Pull request merged**